### DISCUSSION TITLE
PGroonga 2.4.3 リリース!

### DISCUSSION THREAD

**Discussion by komainu8** [pgroonga/pgroonga#283] [Releases]
## PGroonga（ぴーじーるんが） 2.4.3リリース

PostgreSQLで高速日本語全文検索をできるようにする[PGroonga](https://pgroonga.github.io/ja/)の2.4.3をリリースしました！

### 改良

[`pgroonga_highlight_html` 関数](https://pgroonga.github.io/ja/reference/functions/pgroonga-highlight-html.html) ハイライト対象に ``text[]`` 型を指定できるようにしました。 [GitHub#219][Reported by Kentaro Hayashi]

### 修正

  * 2フェーズコミット中にPGroongaのインデックス作成に失敗する問題を修正しました。 [GitHub#269][Reported by Raif Atef]

### [`pgroonga_highlight_html` 関数](https://pgroonga.github.io/ja/reference/functions/pgroonga-highlight-html.html) ハイライト対象に ``text[]`` 型を指定できるようにしました。

  通常、 ``&@~`` を使って ``text[]`` 型のカラムを検索できます。これは、 ``&@~`` が ``text[]`` 型をサポートしているからですが、 ``&@~`` でマッチした結果は ``pgroonga_highlight_html()`` でハイライトできませんでした。 ``pgroonga_highlight_html()`` は ``text[]`` 型をサポートしていないためですが、これは不便です。

  そのため、今回のリリースから以下のように ``&@~`` でマッチした結果を ``pgroonga_highlight_html()`` でハイライトできるようにしました。

  ```sql
  CREATE TABLE memos (
    contents text[]
  );

  CREATE INDEX pgroonga_contents_index
            ON memos
         USING pgroonga(contents);

  INSERT INTO memos VALUES (ARRAY['Debian', 'Ubuntu']);
  INSERT INTO memos VALUES (ARRAY['CentOS', 'AlmaLinux', 'Rocky Linux']);

  SELECT pgroonga_highlight_html(contents, ARRAY['debian'])
    FROM memos
   WHERE contents &@~ 'debian';

               pgroonga_highlight_html              
  --------------------------------------------------
   {"<span class=\"keyword\">Debian</span>",Ubuntu}
  (1 row)
  ```

## まとめ

新規ユーザーの方は、[PGroongaについて](https://groonga.org/ja/blog/2017/10/10/pgroonga-2.0.2.html#pgroonga)も参照してください。

PostgreSQLで高速に日本語全文検索をしたいという方はPGroongaを使ってガンガン検索してください！

**Comment by askdkc** [comment]
今回は既にmacOS用のHomebrewにアップデート出てますね(しかもこのリリースが出るよりも早い😅)

[https://github.com/Homebrew/homebrew-core/pull/120248](https://github.com/Homebrew/homebrew-core/pull/120248)

**Reply by kou** [reply]
Homebrewの人たちは自動で新しいバージョンを検出する仕組みを持っているみたいなんですよねぇ。

**Reply by askdkc** [reply]
> Homebrewの人たちは自動で新しいバージョンを検出する仕組みを持っているみたいなんですよねぇ。

素晴らしいですね👍
前回は何か不調があったんですかね

**Reply by kou** [reply]
新しいバージョンの検出までは自動で、それをpull requestにするのは手動っぽいので、前回はだれも作業しなかっただけなんじゃないですかねぇ。

**Reply by askdkc** [reply]
パッケージの量も多いので抜け漏れちゃうんですかね。気付いたらやっておきます(今回やろうとしたら既にPRがあったので)🫡
